movie.js

```diff
@@ -229,7 +44,198 @@ function calculateTimeSpent() {
 }
 
 document.addEventListener("DOMContentLoaded", () => {
-  displayAllMovies();
+  let movies = [];
+  fetch(
+    "https://raw.githubusercontent.com/bhumikama/bhumikama.github.io/main/movies.json"
+  )
+    .then((response) => {
+      if (!response.ok) {
+        throw new Error("Could not fetch resource");
+      }
+      return response.json();
+    })
+    .then((data) => {
+      movies = data;
+      displayMovies(movies);
+    })
+    .catch((error) => console.error(error));
+
+  function displayMovies(movieList, keyword) {
```

Comment:
It was not necessary to move the definition of your functions `displayMovies`, `createMovieCard`, etc inside the callback of `DOMContentLoaded` event. It actually makes the code harder to read.